pkg/trie/triedb/node_storage.go

```diff
@@ -63,34 +62,34 @@ type (
 	NewStoredNode struct {
 		node Node
 	}
-	CachedStoredNode struct {
+	CachedStoredNode[H any] struct {
```

Comment:
Why not `hash.Hash`?

Reply:
https://github.com/ChainSafe/gossamer/pull/4221#discussion_r1788099883